lib/ecto_ldap/adapter.ex

```diff
@@ -204,11 +206,15 @@ defmodule Ecto.Ldap.Adapter do
     end
   end
 
-  def replace_dn_search_with_uid_present(search_options) when is_list(search_options)do
-    # TODO: This is where we'll pull out the dn term and replace it with
-    # :eldap.present('uid')
-    # We'll then take that extracted dn term and modify the base and the scope
-    search_options
+  def replace_dn_search_with_uid_present(search_terms) when is_list(search_terms) do
+    case search_terms[:filter] do
+      {:and, [and: [equalityMatch: {:AttributeValueAssertion, 'dn', uid}], and: []]} ->
```

Comment:
This feels like a brute-force comparison.  If there is a more graceful way to do this, please let me know!